src/snowflake/snowpark/session.py

```diff
@@ -1568,6 +1568,65 @@ def query_tag(self, tag: str) -> None:
             self._conn.run_query("alter session unset query_tag")
         self._query_tag = tag
 
+    def append_query_tag(self, tag: Union[str, dict], seperator: str = ",") -> None:
+        """
+        Appends a tag to the current query tag.
+        When the input tag is a string the it will be appended to the query tag with a seperator.
+        When the input tag is a dictionary the query tag will be assumed to be a json encoded
+        dictionary. The query tag will be updated with the input tags keys and values.
+
+        Args:
+            tag: The tag to append to the current query tag. When the tag is a string the kj
+            seperator: When appending to a string tag this parameter will be used to seperated values.
+        Note:
+            Assigning a value via session.query_tag will remove any appended query tags.
+
+        Example::
+            >>> session.query_tag = "tag1"
+            >>> session.append_query_tag("tag2")
+            >>> print(session.query_tag)
+            tag1,tag2
+            >>> session.query_tag = "new_tag"
+            >>> print(session.query_tag)
+            new_tag
+
+        Example::
+            >>> session.sql("ALTER SESSION SET QUERY_TAG = 'tag1'").collect()
+            [Row(status='Statement executed successfully.')]
+            >>> session.append_query_tag("tag2")
+            >>> print(session.query_tag)
+            tag1,tag2
+
+        Example::
+            >>> session.query_tag = '{"key1": "value1"}'
+            >>> session.append_query_tag({"key2": "value2"})
+            >>> print(session.query_tag)
+            {"key1": "value1", "key2": "value2"}
+        """
+        if tag:
+            # Ensure local state matches remote
+            self._query_tag = self._conn.run_query("SHOW PARAMETERS LIKE 'QUERY_TAG'")[
+                "data"
+            ][0][1]
+            if isinstance(tag, dict):
+                tag_str = self._query_tag or "{}"
+                try:
+                    tag_dict = json.loads(tag_str)
```

Comment:
I want to point out the fact that you won't be able to call this function twice. Imagine that `self.query_tag = '{"a": 1}'`, then calling `append_query_tag('{"b": 2}')` will adjust `self.query_tag = '{"a": 1}, {"b": 2}'`. Then if you called `append_query_tag('{"c": 3}')` then this line will fail!
If this is the right format then we should not parse `self.query_tag` as it's not a valid JSON

Reply:
That is not how this branch works. For strings it would work that way, but for dicts it's like this since I'm calling `dict.update`.
```
>>> self.query_tag = '{"a": 1}',
>>> append_query_tag('{"b": 2}')
>>> print(self.query_tag)
{"a": 1, "b": 2}
>>> append_query_tag('{"c": 3}')
>>> print(self.query_tag)
{"a": 1, "b": 2, "c": 3}
```

Reply:
I suggest we only support string. Anybody requested us to support dict?

Reply:
Yes, when I talked with the solutions team they said that adding tags as json was a common use case.

Reply:
I think this naming will introduce problems down the line, please see my other comment for introducing a 2nd helper function
```
> session.query_tag = ""
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194bf-0507-768d-000c-a903282f7baf] alter session unset query_tag
> session.append_query_tag({"test": True})
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194bf-0507-768d-000c-a903282fbbff] SHOW PARAMETERS LIKE 'QUERY_TAG'
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194bf-0507-8379-000c-a903282fc113] alter session set query_tag = '{"test": true}'
> session.append_query_tag('{"test2": true}')
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194bf-0507-8379-000c-a90328300707] SHOW PARAMETERS LIKE 'QUERY_TAG'
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194bf-0507-768d-000c-a903283011d7] alter session set query_tag = '{"test": true},{"test2": true}'
> session.query_tag
'{"test": true},{"test2": true}'
```
and plugging in your example:
```
> session.query_tag = '{"a": 1}'
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194c1-0507-768d-000c-a9032831167b] alter session set query_tag = '{"a": 1}'
> session.append_query_tag('{"b": 2}')
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194c1-0507-768d-000c-a90328313f2f] SHOW PARAMETERS LIKE 'QUERY_TAG'
DEBUG    snowflake.snowpark._internal.server_connection:server_connection.py:371 Execute query [queryID: 01b194c1-0507-768d-000c-a90328313f43] alter session set query_tag = '{"a": 1},{"b": 2}'
> session.query_tag
'{"a": 1},{"b": 2}'
```